src/os/bluestore/BlockDevice.cc

```diff
@@ -464,6 +487,37 @@ int BlockDevice::read(uint64_t off, uint64_t len, bufferlist *pbl,
   return r < 0 ? r : 0;
 }
 
+int BlockDevice::read_buffered(uint64_t off, uint64_t len, char *buf)
```

Comment:
This doesn't enforce block-alignment, it is always buffered (uses page 
cache), and it doesn't allocate any memory.  This is the path taken by 
rocksdb when reading small bits of the sst files (which it doesn't 
explicitly cache--it relies on the page cache).